packages/stark-ui/src/modules/transform-input-directive/directives/transform-input.directive.ts

```diff
@@ -0,0 +1,141 @@
+ *
+ * example:
+ * ```
+ * <input [(ngModel)] StarkTransformInputDirective="uppercase" />
+ * ```
+ */
+@Directive({
+	// tslint:disable-next-line:directive-selector
+	selector: "[starkTransformInput]",
+	providers: [STARK_TRANSFORM_INPUT_PROVIDER],
+	host: {
+		"(input)": "_onInput($event)",
+		"(blur)": "_onTouched()"
+	}
+})
+export class StarkTransformInputDirective implements ControlValueAccessor, OnChanges {
+	public _transformation: (value: any) => any;
+
+	// tslint:disable-next-line:no-input-rename
+	@Input("starkTransformInput")
+	public set transformation(transformation: StarkInputTransformationType) {
+		switch (transformation) {
+			case UPPERCASE:
+				this._transformation = (v: string) => v.toUpperCase();
+				break;
+			case LOWERCASE:
+				this._transformation = (v: string) => v.toLocaleLowerCase();
+				break;
+			default:
+				this._transformation = transformation;
+				break;
+		}
+	}
+
+	/**
+	 * @ignore
+	 * The registered callback function called when an input event occurs on the input element.
+	 */
+	private _onChange: (_: any) => void = (_: any) => {
+		/*noop*/
+	};
+
+	/**
+	 * @ignore
+	 * The registered callback function called when a blur event occurs on the input element.
+	 */
+	public _onTouched: () => void = () => {
+		/*noop*/
+	};
+
+	/**
+	 * Class constructor
+	 * @param _renderer - Angular renderer
+	 * @param _elementRef - Reference to the element
+	 */
+	public constructor(private _renderer: Renderer2, private _elementRef: ElementRef) {}
+
+	/**
+	 * Sets the "value" property on the input element.
+	 *
+	 * @param value The checked value
+	 */
+	public writeValue(value: any): void {
+		const normalizedValue: any = value === null ? "" : value;
+		this._renderer.setProperty(this._elementRef.nativeElement, "value", normalizedValue);
+	}
+
+	/**
+	 * Registers a function called when the control value changes.
+	 *
+	 * @param fn The callback function
+	 */
+	public registerOnChange(fn: (_: any) => void): void {
+		this._onChange = fn;
+	}
+
+	/**
+	 * Registers a function called when the control is touched.
+	 *
+	 * @param fn The callback function
+	 */
+	public registerOnTouched(fn: () => void): void {
+		this._onTouched = fn;
+	}
+
+	/**
+	 * Sets the "disabled" property on the input element.
+	 *
+	 * @param isDisabled The disabled value
+	 */
+	public setDisabledState(isDisabled: boolean): void {
+		this._renderer.setProperty(this._elementRef.nativeElement, "disabled", isDisabled);
+	}
+
+	/**
+	 * @ignore
+	 * Listens to input event
+	 */
+	public _onInput(event: Event): void {
+		const value: any = (<HTMLInputElement>event.target).value;
+		const transformed: any = this._transformation(value);
+		if (transformed !== value) {
+			this._elementRef.nativeElement.value = transformed;
+			this._onChange(transformed);
+		} else {
+			this._onChange(value);
+		}
+	}
+
+	/**
+	 * Angular life cycle hook
+	 * @param changes - simple changes
+	 */
+	public ngOnChanges(changes: SimpleChanges): void {
```

Comment:
Could you move this `ngOnChanges` up right after the constructor? So it can be easily spotted